### DISCUSSION TITLE
RDF4J in appinventor

### DISCUSSION THREAD

**Discussion by Antus8** [eclipse-rdf4j/rdf4j#3551] [Q&A]
I'm trying to integrate RDF4J library in an AppInventor project. Every time I get the following error:
```
 org.eclipse.rdf4j.rio.UnsupportedRDFormatException: Did not recognise RDF format object Turtle (mimeTypes=text/turtle, application/x-turtle; ext=ttl)
08-29 12:27:10.417 30454 30499 W System.err:     at java.util.Optional.orElseThrow(Optional.java:285)
08-29 12:27:10.417 30454 30499 W System.err:     at org.eclipse.rdf4j.rio.Rio.createWriter(Unknown Source)
```
Even if I use the Maven ServicesResourceTransformer shade plugin to compile my code without losing parsers I get this error. I read that AppInventor uses Java 6, so I think there is a problem related to Java versions incompatibility. Can someone help me with this problem? Thanks

**Comment by barthanssens** [comment]
Hmz, have you explicitly included the  rdf4j-rio-turtle.jar in the project ? If that jar is not present it will produce an UnsupportedFormatException...

And RDF4J 3.x is indeed only tested/supported on Java 8 and newer ...

**Reply by Antus8** [reply]
I build it using maven and it is included as a dependency..
Also I downgraded to RDF4J 1.0.x because I had problems with lambda expressions not supported

**Reply by Antus8** [reply]
I read that RDF4J relies on META-INF/services that are provided in the jar file after maven building process. But I think that the folder is not included in Android APKs. Is this correct? If so, how can I solve this problem?

**Comment by barthanssens** [comment]
No idea to be honest, sorry, I don't know anything about development on Android...

There once was an effort to port RDF4J 2.x to Android (https://github.com/MM2-0/rdf4a). It is not actively being developed anymore, but perhaps that might help you with your project...
